libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2142,9 +2142,14 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 					" outside of storage."
 				);
 		}
+
+		string functionNameSuggestion =".";
+		if(getNameSuggestion(*m_scope, memberName, functionNameSuggestion))
```

Comment:
Instead of passing the suggestion by non-const reference, setting it in the function and then using a ``bool`` to signal if there's a suggestion or not, ``getNameSuggestion`` could return the suggestion right away or an empty string if there's non.

Reply:
for this change I would need to do something like this.
```
string suggestion = getNameSuggestion(contract, _missingFunctionName,false);
if (!suggestion.empty())
  {
  if (_first)
     suggestion = decl->name() + "." + suggestion;
  else
    suggestion = decl->name() + "()." + _suggestion;
  return suggestion;
 }
```

It make sense to make this change?

Reply:
@chriseth Should I implement this change?